lib/transaction.js

```diff
@@ -48,60 +50,12 @@ class Transaction extends EventEmitter {
     //      2 separate Promise Chains.  This ensures that each Promise Chain
     //      can establish its error-handling semantics without interfering
     //      with the other Promise Chain.
-    const basePromise = this.acquireConnection(config, (connection) => {
-      const trxClient = (this.trxClient = makeTxClient(
-        this,
-        client,
-        connection
-      ));
-      const init = client.transacting
-        ? this.savepoint(connection)
-        : this.begin(connection);
-      const executionPromise = new Bluebird((resolver, rejecter) => {
-        this._resolver = resolver;
-        this._rejecter = rejecter;
-      });
-
-      init
-        .then(() => {
-          return makeTransactor(this, connection, trxClient);
-        })
-        .then((transactor) => {
-          transactor.executionPromise = executionPromise;
-
-          // If we've returned a "thenable" from the transaction container, assume
-          // the rollback and commit are chained to this object's success / failure.
-          // Directly thrown errors are treated as automatic rollbacks.
-          let result;
-          try {
-            result = container(transactor);
-          } catch (err) {
-            result = Bluebird.reject(err);
-          }
-          if (result && result.then && typeof result.then === 'function') {
-            result
-              .then((val) => {
-                return transactor.commit(val);
-              })
-              .catch((err) => {
-                return transactor.rollback(err);
-              });
-          }
-          return null;
-        })
-        .catch((e) => {
-          return this._rejecter(e);
-        });
-
-      return executionPromise;
-    });
-
+    const basePromise = this._kickThingsOff(config, container);
 
     // FYI: This is the Promise Chain for EXTERNAL use.  It ensures that the
     //      caller must handle any exceptions that result from `basePromise`.
-    this._promise = basePromise.then((x)=> x);
+    this._promise = Bluebird.resolve(basePromise).then((x)=> x);
```

Comment:
Added `Bluebird.resolve(..)` as a temporary work-around since various portions of the code still rely upon the `.finally(..)` method.